### ISSUE TITLE
What the `ddim_step` does in the Algorithm 2 of the paper

### ISSUE THREAD

**Issue by GT9505** [ShoufaChen/DiffusionDet#7]
Hi~, I wonder to know what the DDIM does in the `ddim_step` of Algorithm 2 of the paper.
Could you describe intuitionally how the DDIM changes the `pred_bboxes`.
I don't get why DDIM can improve the bbox AP.

**Comment by tyshiwo1** [comment]
...same question

**Comment by ShoufaChen** [comment]
Hi, thanks for your interests in our work.



DDIM [a] is a variant of DDPM [b] with the goal of accelerating sampling process while maintaining performance. It is used to generate the input of next sampling step from the output of current step. More specifically, `DDIM` is implemented [here](https://github.com/ShoufaChen/DiffusionDet/blob/main/diffusiondet/detector.py#L224-L234) in our code.

Intuitively, DiffusionDet is trained with noisy boxes, while the output of sampling step is the prediction of the network, instead of 'noise'. Therefore, DDIM is used to transform the prediction to 'noisy distribution'.

[a] Song, Jiaming, Chenlin Meng, and Stefano Ermon. "Denoising diffusion implicit models." ICLR 2021.

[b] Ho, Jonathan, Ajay Jain, and Pieter Abbeel. "Denoising diffusion probabilistic models." NeurIPS 2020.

**Comment by tyshiwo1** [comment]
Thank you for your reply.

> Hi, thanks for your interests in our work.
> 
> DDIM [a] is a variant of DDPM [b] with the goal of accelerating sampling process while maintaining performance. It is used to generate the input of next sampling step from the output of current step. More specifically, `DDIM` is implemented [here](https://github.com/ShoufaChen/DiffusionDet/blob/main/diffusiondet/detector.py#L224-L234) in our code.
> 
> Intuitively, DiffusionDet is trained with noisy boxes, while the output of sampling step is the prediction of the network, instead of 'noise'. Therefore, DDIM is used to transform the prediction to 'noisy distribution'.
> 
> [a] Song, Jiaming, Chenlin Meng, and Stefano Ermon. "Denoising diffusion implicit models." ICLR 2021.
> 
> [b] Ho, Jonathan, Ajay Jain, and Pieter Abbeel. "Denoising diffusion probabilistic models." NeurIPS 2020.

**Comment by huilicici** [comment]
Hi, Thank you for your explanation. In DDIM or DDPM, there are losses (KL-Divergence) to constrain the diffused outputs during training steps to be Gaussian distributions. I thought it is the base for DDIM sampling (the reverse process). However, in DiffusionDet, only set prediction loss is used. I still cannot get why DDIM can work without training constrain.

**Comment by ShoufaChen** [comment]
Hi, @huilicici 

Thanks for your interest. Would you mind opening a new issue to discuss your concern about the loss since it is out the scope of this issue.

**Comment by huilicici** [comment]
> Hi, @huilicici
> 
> Thanks for your interest. Would you mind opening a new issue to discuss your concern about the loss since it is out the scope of this issue.

All right. Thank you.
